### PR TITLE
[WIP] Dry-run mode

### PR DESCRIPTION
**Pull request by DavertMik**
In this mode, you can print tests output with no commands actually being executed.
This mode disables all helpers and plugins but preserves all events and console output, so you can easily see what all tests are doing, and validate your tests before running them completely

### Usage

```
npx codeceptjs dry-run
```

**OUTPUT**

Screenshot from 2019-08-06 02-44-30

When passed `--steps` or `--debug` option runs tests, disabling all plugins and helpers, so you can get step-by-step report with no tests actually executed.

```
npx codeceptjs dry-run --steps
```

If a plugin needs to be enabled in `dry-run` mode, pass its name in `-p` option:

```
npx codeceptjs dry-run --steps -p allure
```

**OUTPUT**

Screenshot from 2019-08-05 12-04-51

### CODE REVIEW

**Comment by pablopaul** [comment]
@DavertMik what is the use case for dry mode?

**Comment by DavertMik** [comment]
@pablopaul Use cases:

- To print all test scenarios in a human-readable format on a screen
- Create reports for tests without running them
- Check actual steps will be executed when a test will be run.
- Dry run mode can also be used as validator for  JS tests (validate all tests before running them), however, it has limitations working with `grep*` so I'm not sure about this use-case. However, if all custom code is stored in helpers and not used in tests, such validation will pass

**Comment by DavertMik** [comment]
Btw, the similar PR for mocha was never merged https://github.com/mochajs/mocha/pull/1070

File: docs/commands.md [hunk 1]
```diff
@@ -13,152 +13,178 @@ Executes tests. Requires `codecept.conf.js` config to be present in provided pat
...
+If a plugin needs to be enabled in `dry-run` mode, pass its name in `-p` option:
+
+```
+npx codeceptjs dry-run --steps -p allure
+```
+
+To enable bootstrap script in dry-run mode, pass in `--bootstrap` option when running with `--steps` or `--debug`
+
+```
+nox codeceptjs dry-run --steps --bootstrap
```
Comment (nabeelharri) [inline docs/commands.md]: npx instead of nox

Comment (DavertMik) [reply docs/commands.md]: yes, thank you. Could you make a pull request fixing this?

Comment (nabeelharri) [reply docs/commands.md]: Sure

**Pull request merged**